### PR TITLE
Recovery API

### PR DESCRIPTION
**Pull request by ch3njust1n**
# Description

> Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Checklist

- [x] This PR can be reviewed in under 30 minutes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have assigned reviewers to this PR.

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Environment variables declared in this file are automatically made available to Prisma.
+# See the documentation for more detail: https://pris.ly/d/prisma-schema#accessing-environment-variables-from-the-schema
+
+# Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB and CockroachDB.
+# See the documentation for all the connection string options: https://pris.ly/d/connection-strings
+
+DATABASE_URL="postgres://justin@localhost:5432/postgres"
```
Comment (cbolles) [inline .env]: Looks like a regression of the `.env` being added

File: packages/server/package.json [hunk 2]
```diff
@@ -31,16 +31,19 @@
...
     "class-validator": "^0.13.2",
     "crypto-js": "^4.1.1",
     "date-fns": "^2.29.3",
     "dotenv": "^16.0.3",
     "joi": "^17.7.0",
+    "nodemailer": "^6.8.0",
     "passport": "^0.6.0",
     "passport-jwt": "^4.0.0",
     "passport-local": "^1.0.0",
+    "prisma": "^4.5.0",
```
Comment (wenhwang97) [inline packages/server/package.json]: Do we need to have this in dependencies? @gregfrasco @cbolles

File: packages/server/src/auth/auth.controller.ts [hunk 3]
```diff
@@ -30,3 +30,18 @@ export class SignupController {
...
+export class RecoveryController {
+  constructor(private authService: AuthService) {}
+
+  @Post('forgot')
+  async forgotPassword(@Body('projectId') projectId: string, @Body('email') email: string): Promise<void> {
+    return this.authService.forgotPassword(projectId, email);
+  }
+
+  @Post('reset')
+  async resetPassword(@Body('projectId') projectId: string, @Body('email') email: string, @Body('password') password: string, @Body('resetCode') resetCode: string): Promise<void> {
```
Comment (cbolles) [inline packages/server/src/auth/auth.controller.ts]: Suggestion: May want to make a DTO for the reset

**Review by cbolles — CHANGES_REQUESTED** [review]
Looks good, just looks like some changes were overwritten and an optional suggestion

**Review by wenhwang97 — CHANGES_REQUESTED** [review]
One question, everything else looks good

File: packages/server/package.json (hunk 2 above)
Comment (ch3njust1n) [reply packages/server/package.json]: removed in latest commit

File: packages/server/src/auth/auth.controller.ts (hunk 3 above)
Comment (ch3njust1n) [reply packages/server/src/auth/auth.controller.ts]: Good suggestion. Added in latest commit.

**Review by cbolles — DISMISSED** [review]

**Review by wenhwang97 — DISMISSED** [review]

**Review by cbolles — DISMISSED** [review]

**Review by gregfrasco — DISMISSED** [review]

**Review by gregfrasco — APPROVED** [review]

**Pull request merged**